src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+
+    private int size;
+
+    private Object[] dataArray;
+
+    public ArrayList() {
+        this.dataArray = new Object[START_CAPACITY];
+        this.size = 0;
+    }
+
+    private boolean sizeCheck() {
```

Comment:
The method name `sizeCheck` is not informative enough to indicate what the method does. Consider renaming it to something more descriptive, such as `isArrayFull`, which would make it clear that it checks if the array has reached its capacity.